### ISSUE TITLE
session restore: "q" at swap dialog loads file instead of (Q)uitting

### ISSUE THREAD

**Issue by emberfade** [neovim/neovim#12213]
- `nvim --version`: NVIM v0.4.3
- `vim -u DEFAULTS` (version: VIM - Vi IMproved 8.2) behaves differently? Yes
- Operating system/version: Arch Linux
- Terminal name/version: alacritty 0.4.2
- `$TERM`: alacritty

### Steps to reproduce using `nvim -u NORC`

```
touch file
nvim -u NORC file
:mksession nvim.session
```
Keep the file opened.

In a second terminal:
```
nvim -u NORC -S nvim.session
q<CR>
```

### Actual behaviour

In the second terminal the error about an existing swap file shows up.  After pressing `q<CR>` to quit and continue, it displays `file` in the window.

### Expected behaviour

After pressing `q<CR>` to quit, the window should have shown an empty buffer instead of `file`.  `:buffers` lists the empty buffer and additionally `file` which is inactive.

If I do the following in the second terminal
```
nvim -u NORC
:e file
q
```
it *does not* want me to press enter and instead puts `file` in the buffer list without displaying it.  This is what I would have expected when loading a session as well.

edit: Rewrite actual and expected behaviour.

**Comment by bodgergely** [comment]
I failed to reproduce the bug with your steps on v0.4.3. For me the usual swap file exists warning comes up.

**Comment by emberfade** [comment]
My description about actual and expected behaviour was not well done.  Therefore I edited those sections in the original post.

The error about the existing swap file does show up.  But it then should not display the `file` when I choose `q` to quit.

**Comment by justinmk** [comment]
Thanks for the report. Can you try the development version? 

- The [Releases](https://github.com/neovim/neovim/releases/tag/nightly) page has pre-built archives for Linux/Windows/macOS. 
- Check `:version` to confirm that you are using the latest version.

**Comment by emberfade** [comment]
* `:version`: NVIM v0.5.0-478-g42b441738

Happens with it as well.

**Comment by justinmk** [comment]
confirmed.

**Comment by bashdev** [comment]
Think I found  the issue.
While closing the buffer the close_buffer function was called to unload  the buffer the DOBUF_UNLOAD macro.

```c
    close_buffer(curwin, curbuf, DOBUF_UNLOAD, false);
```
changing it to DOBUF_DEL seems to solve the issue
```c
    close_buffer(curwin, curbuf, DOBUF_DEL, false);
```
I might be wrong, this is my first time with a open source project.

**Comment by Alaskra** [comment]
In nvim v0.7.0, I can't confirm it. Maybe have been solved?

**Comment by justinmk** [comment]
Thanks @Alaskra ! Confirmed fixed in b7d70de345db
